src/PostgREST/App.hs

```diff
@@ -358,7 +358,9 @@ handleInfo identifier RequestContext{..} =
     allOrigins = ("Access-Control-Allow-Origin", "*")
     allowH table =
       ( HTTP.hAllow
-      , if tableInsertable table then "GET,POST,PATCH,DELETE" else "GET"
+      , "GET" <> (if tableInsertable table then ",POST" else "")
+              <> (if tableUpdatable table then ",PATCH" else "")
+              <> (if tableDeletable table then ",DELETE" else "")
```

Comment:
How about:
```haskell
intercalate "," $
   ["GET"] ++ ["POST" | tableInsertable] ++ ["PATCH" | tableUpdatable] ++ ["DELETE" | tableDeletable]
```

Reply:
Looks cleaner, I'll give it a try.

Reply:
How are we going to handle `PUT`? Should we just say `insertable + updatable = PUT`?

Reply:
Actually.. we'd need the presence of a PK, too, right?

Reply:
And... for completeness, we should probably add `HEAD` and `OPTIONS` by default, as well!